### PR TITLE
add github actions generating

### PR DESCRIPTION
**Pull request by wdpypere**

### CODE REVIEW

File: lib/vsc/install/ci.py [hunk 1]
```diff
@@ -62,19 +65,68 @@
...
         raise IOError("Failed to write %s: %s" % (path, err))
 
+def gen_github_action():
+    """
+    Generate tox.ini configuration file for github actions.
+    """
+    logging.info('[%s]', GITHUB_ACTIONS)
+    vsc_ci_cfg = parse_vsc_ci_cfg()
+
+    name_url = "https://github.com/hpcugent/vsc-install"
```
Comment (wdpypere) [inline lib/vsc/install/ci.py:92]: this won't work yet as I need to get this line dynamically for each repo, I tried the code of headers and it does not work. @boegel suggestions?

File: lib/vsc/install/ci.py [hunk 2]
```diff
@@ -62,19 +65,68 @@
...
+            "DO NOT EDIT MANUALLY",
+        ]
+
+        txt = ['# ' + l for l in header]
+        yaml_content = {
+            'name': 'run python tests',
+            'on': ['push', 'pull_request'],
+            'jobs': {
+                'python_unittests': {
+                    'runs-on': 'ubuntu-20.04',
```
Comment (wpoely86) [inline lib/vsc/install/ci.py:107]: Maybe directly run it inside an rhel 8 container? That's our production environment anyway.

File: lib/vsc/install/ci.py (hunk 1 above)
Comment (wdpypere) [reply lib/vsc/install/ci.py:92]: I just noticed the headers script is used with `REPO_BASE_DIR=$PWD`, maybe this will require the same trick.

Comment (wdpypere) [reply lib/vsc/install/ci.py]: well, found a way to fix it, but I have to admit I don't understand how it works.

File: lib/vsc/install/ci.py (hunk 2 above)
Comment (wdpypere) [reply lib/vsc/install/ci.py:109]: according to https://github.com/actions/runner-images there is no rhel 8 container.

**Comment by wdpypere** [comment]
fixes #214

File: lib/vsc/install/ci.py [hunk 3]
```diff
@@ -62,19 +65,70 @@
...
 RUN_SHELLCHECK = 'run_shellcheck'
+ENABLE_GITHUB_ACTIONS = 'enable_github_actions'
 
 logging.basicConfig(format="%(message)s", level=logging.INFO)
 
 
 def write_file(path, txt):
     """Write specified contents to specified path."""
+
+    if not os.path.exists(os.path.dirname(path)):
```
Comment (stdweird) [inline lib/vsc/install/ci.py:76]: use `exist_ok = True` in makedirs instead of the test and the copy/pasting

File: lib/vsc/install/ci.py [hunk 4]
```diff
@@ -62,19 +65,70 @@
...
+                            'python': [3.6, 3.9]
+                        }
+                    },
+                    'steps': [
+                        {'name': 'Checkout code', 'uses': 'actions/checkout@v3'},
+                        {'name': 'Setup Python', 'uses': 'actions/setup-python@v4',
+                         'with': {'python-version': '${{ matrix.python }}'}},
+                        {'name': 'install tox', 'run': 'pip install tox'},
+                        {'name': 'add mandatory git remote',
+                         'run': 'git remote add hpcugent %s.git' % name_url},
```
Comment (stdweird) [inline lib/vsc/install/ci.py:121]: ue f string

**Comment by stdweird** [comment]
minor remakr, nice work @wdpypere

File: lib/vsc/install/ci.py (hunk 4 above)
Comment (wdpypere) [reply lib/vsc/install/ci.py:120]: I will, as soon as this package is python 3 only :wink:

**Comment by wdpypere** [comment]
@stdweird remarks addressed

**Review by boegel — APPROVED** [review]

File: lib/vsc/install/ci.py (hunk 2 above)
Comment (wpoely86) [reply lib/vsc/install/ci.py:108]: Not a runner image, a container. Just add a line like: 
```
    runs-on: ubuntu-20.04
    container: rockylinux:8
    steps:
```
Or whatever your favorite rhel flavour is.

**Pull request merged**